apps/nestjs-indexer/src/services/candlestick/candlesticks.service.ts

```diff
@@ -0,0 +1,126 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { PrismaService } from 'src/prisma/prisma.service';
+
+@Injectable()
+export class CandlestickService {
+  private readonly logger = new Logger(CandlestickService.name);
+
+  constructor(private readonly prismaService: PrismaService) {}
+
+  /**
+   * Generate candlestick data for a specific token and time interval.
+   * @param tokenAddress - The token address.
+   * @param intervalMinutes - The time interval in minutes (e.g., 1, 5, 15, 60).
+   */
+  async generateCandles(tokenAddress: string, intervalMinutes: number) {
```

Comment:
We have already something in the apps/data-backend to generate a graph simulated from it.

todo: also to have an endpoint in the apps/data-backend to render this graph per token, starting with 5m candle by default.

Also if we do and save it in the indexer, we need to not wait this service as explained above.